enterprise/frontend/src/metabase-enterprise/sandboxes/components/LoginAttributesWidget/LoginAttributesWidget.tsx

```diff
@@ -0,0 +1,103 @@
+import { useField } from "formik";
+import { type HTMLAttributes, useMemo } from "react";
+import { t } from "ttag";
+
+import { skipToken, useGetUserQuery } from "metabase/api";
+import FormField from "metabase/common/components/FormField";
+import { Accordion, Box, Loader, Text } from "metabase/ui";
+import { useGetTenantQuery } from "metabase-enterprise/api";
+import { getExtraAttributes } from "metabase-enterprise/sandboxes/utils";
+import type {
+  StructuredUserAttributes,
+  UserAttributeKey,
+  UserAttributeMap,
+  UserAttributeValue,
+  UserId,
+} from "metabase-types/api";
+
+import { LoginAttributeMappingEditor } from "./LoginAttributeMappingEditor";
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  name: string;
+  title?: string;
+  description?: string;
+  userId?: UserId;
+}
+
+// if the value set in the ui is the same as the tenant value, don't save it to the user
+const isTenantValue = (
+  [key, inputValue]: [UserAttributeKey, UserAttributeValue],
+  structuredAttributes: StructuredUserAttributes,
+) => {
+  const attribute = structuredAttributes[key];
+  if (!attribute) {
+    return false;
+  }
+
+  const tenantValue =
+    attribute?.original?.value ??
+    (attribute.source === "tenant" ? attribute.value : undefined);
+  return tenantValue === inputValue;
+};
+
+export const LoginAttributesWidget = ({
+  name = "login_attributes",
+  title = t`Attributes`,
+  description,
+  className,
+  style,
+  userId,
+}: Props) => {
+  const [{ value }, , { setValue, setError }] = useField(name);
+  const [{ value: tenantId }] = useField("tenant_id");
+
+  const { data: userData, isLoading } = useGetUserQuery(userId ?? skipToken);
```

Comment:
I don't love that we're reaching out to RTK Query here, but it's probably not a big deal. If there is a way to pass thing through the form context, I'd prefer that though

Reply:
I don't love it either, but passing non-form-value data through form context sounds even worse 😞